titus-server-master/src/test/java/com/netflix/titus/master/integration/v3/job/JobEbsVolumesTest.java

```diff
@@ -0,0 +1,223 @@
+        List<EbsVolume> ebsVolumes = serviceJobDescriptor.getContainer().getContainerResources().getEbsVolumes();
+
+        // String volumeId = ONE_TASK_SERVICE_JOB.getContainer().getContainerResources().
+        jobsScenarioBuilder.schedule(serviceJobDescriptor, jobScenarioBuilder ->
+                jobScenarioBuilder
+                        // Start the initial tasks
+                        .template(ScenarioTemplates.startTasksInNewJob())
+                        .expectSome(1, taskScenarioBuilder -> taskScenarioBuilder.getTask().getTaskContext().getOrDefault(TaskAttributes.TASK_ATTRIBUTES_EBS_VOLUME_ID, "").equals(ebsVolumes.get(0).getVolumeId()))
+                        .expectSome(1, taskScenarioBuilder -> taskScenarioBuilder.getTask().getTaskContext().getOrDefault(TaskAttributes.TASK_ATTRIBUTES_EBS_VOLUME_ID, "").equals(ebsVolumes.get(1).getVolumeId()))
+                        .expectSome(1, taskScenarioBuilder -> taskScenarioBuilder.getTask().getTaskContext().getOrDefault(TaskAttributes.TASK_ATTRIBUTES_EBS_VOLUME_ID, "").equals(ebsVolumes.get(2).getVolumeId()))
+                        // Finish the initial tasks and make sure they are replaced
+                        .inTask(0, taskScenarioBuilder -> taskScenarioBuilder.transitionUntil(TaskStatus.TaskState.Finished))
+                        .inTask(0, taskScenarioBuilder -> taskScenarioBuilder.expectStateUpdateSkipOther(TaskStatus.TaskState.Finished))
+                        .inTask(1, taskScenarioBuilder -> taskScenarioBuilder.transitionUntil(TaskStatus.TaskState.Finished))
+                        .inTask(1, taskScenarioBuilder -> taskScenarioBuilder.expectStateUpdateSkipOther(TaskStatus.TaskState.Finished))
+                        .inTask(2, taskScenarioBuilder -> taskScenarioBuilder.transitionUntil(TaskStatus.TaskState.Finished))
+                        .inTask(2, taskScenarioBuilder -> taskScenarioBuilder.expectStateUpdateSkipOther(TaskStatus.TaskState.Finished))
+                        .expectAllTasksCreated()
+                        .allTasks(TaskScenarioBuilder::expectTaskOnAgent)
+                        .assertTasks(task -> task.get(0).getResubmitNumber() == 1)
+                        .assertTasks(task -> task.get(1).getResubmitNumber() == 1)
+                        .assertTasks(task -> task.get(2).getResubmitNumber() == 1)
+                        .inTask(0, taskScenarioBuilder -> taskScenarioBuilder.template(ScenarioTemplates.startTask()))
+                        .inTask(1, taskScenarioBuilder -> taskScenarioBuilder.template(ScenarioTemplates.startTask()))
+                        .inTask(2, taskScenarioBuilder -> taskScenarioBuilder.template(ScenarioTemplates.startTask()))
+                        // Make sure replacements have correct attribute
+                        .expectSome(1, taskScenarioBuilder -> taskScenarioBuilder.getTask().getTaskContext().getOrDefault(TaskAttributes.TASK_ATTRIBUTES_EBS_VOLUME_ID, "").equals(ebsVolumes.get(0).getVolumeId()))
```

Comment:
How ebs volume / task replacement checked here? Is it possible that ebs volumes might be assigned here in a different way, and the test still would pass?